Fix Crafting Pattern Models

pauljoda opened the pull request:
This will standardize the transforms to look like a normal item and will also apply the IItemColor object from the stored item to the item, allowing things like potions, spawn eggs, and other implementations to show correct color

pauljoda committed:
Fix perspective issues

pauljoda committed:
Fix color for child models

raoulvdberge commented on src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java:
```diff
@@ -257,6 +263,26 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
+            public int getColorFromItemstack(ItemStack stack, int tintIndex) {
+                CraftingPattern pattern = ItemPattern.getPatternFromCache(Minecraft.getMinecraft().world, stack);
+
+                if (GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1 &&
```
Can we move the `GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1` check to a static method in `BakedModelPattern` and reuse it there as well?

raoulvdberge commented:
Thanks, left one small remark.

raoulvdberge commented on src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java:
```diff
@@ -257,6 +263,26 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
+                CraftingPattern pattern = ItemPattern.getPatternFromCache(Minecraft.getMinecraft().world, stack);
+
+                if (GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1 &&
+                        Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex) != -1) {
```
`Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex)` can probably go in another variable, we're reusing it twice.

pauljoda replied:
Addressed in last commit

pauljoda committed:
To address raoulvberge's remarks

Adds static method for rendering pattern, holds local variable for
colors object

raoulvdberge commented:
Thank you, merged.
